### ISSUE TITLE
Incorrect best score and best hyper parem

### ISSUE THREAD

**Issue by stramzik** [keras-team/keras-tuner#656]
**Describe the bug**

I used a keras-tuner to search for best hyperparameters on model with 4 different variables. The objective was for the val_loss to go down and below is the summary of tuner search

`
Results summary
Results in BiLSTM_A\BiLSTM_Trial
Showing 10 best trials
Objective(name='val_loss', direction='min')
Trial summary
Hyperparameters:
out_dim: 20
units: 100
activation_func: softmax
learning_rate: 0.01
Score: 0.055424764752388
Trial summary
Hyperparameters:
out_dim: 60
units: 100
activation_func: relu
learning_rate: 0.01
Score: nan
Trial summary
Hyperparameters:
out_dim: 40
units: 75
activation_func: softmax
learning_rate: 0.01
Score: 0.054255176335573196
Trial summary
Hyperparameters:
out_dim: 20
units: 75
activation_func: softmax
learning_rate: 0.01
Score: 0.054356783628463745
Trial summary
Hyperparameters:
out_dim: 60
units: 50
activation_func: softmax
learning_rate: 0.01
Score: 0.054930560290813446
Trial summary
Hyperparameters:
out_dim: 40
units: 50
activation_func: softmax
learning_rate: 0.01
Score: 0.056117504835128784
Trial summary
Hyperparameters:
out_dim: 60
units: 50
activation_func: softmax
learning_rate: 0.001
Score: 0.05824028328061104
Trial summary
Hyperparameters:
out_dim: 60
units: 50
activation_func: softmax
learning_rate: 0.0001
Score: 0.11464226990938187
Trial summary
Hyperparameters:
out_dim: 40
units: 75
activation_func: softmax
learning_rate: 0.0001
Score: 0.1221684068441391
Trial summary
Hyperparameters:
out_dim: 40
units: 75
activation_func: relu
learning_rate: 0.01
Score: nan
`

And according to the above summary the best parameters should be as below 
```
out_dim: 40
units: 75
activation_func: softmax
learning_rate: 0.01
```

where as when i use tuner.get_best_hyperparameters()[0] this o/p the best params 
```
out_dim: 20
units: 100
activation_func: softmax
learning_rate: 0.01

```
why is that?

**To Reproduce**

**Expected behavior**

**Additional context**

**Would you like to help us fix it?**

**Comment by haifeng-jin** [comment]
It seems the trials are not correctly sorted. When do `results_summary` it should put the best result first.
I cannot reproduce the error with our example code. If anyone have the same issue, it is much appreciated if can provide a short code snippet to reproduce the bug.

**Comment by schmitzn** [comment]
I have the same issue. I think it occurs when there are trials with score value `nan` .
It seems that the `sorted()` function cannot handle `nan` properly:

```
>>> import numpy as np
>>> a = [2,3,4,np.nan,5,6,7,np.nan, 1, np.nan,8]
>>> sorted(a)
[2, 3, 4, nan, 5, 6, 7, nan, 1, nan, 8]
```

I guess the same happens when sorting in the oracle class:

https://github.com/keras-team/keras-tuner/blob/98e25d255d53ad56296fdb35327b6201cabaf35e/keras_tuner/engine/oracle.py#L305-L310

**Comment by haifeng-jin** [comment]
This will be solved along with #704 .
The nan trials will be reported as invalid.
